synapse/event_auth.py

```diff
@@ -137,6 +137,22 @@ def check_auth_rules_for_event(
     Raises:
         AuthError if the checks fail
     """
+    # Before we get too far into event auth, validate that the event is even
+    # valid enough to be used
+    if event.type == EventTypes.PowerLevels:
```

Comment:
this might be in the wrong place, but it's highly convenient to put it here

Reply:
It seems like for similar code we usually do it in `EventValidator.validate_new` + `event_from_pdu_json`, but I'm unsure if that makes sense or not. (I was looking at where we call `validate_canonicaljson`)

Reply:
We should really add a function to the `Validator` for inbound federation events, I guess